server/src/main/java/io/crate/metadata/cluster/AlterTableClusterStateExecutor.java

```diff
@@ -69,9 +67,6 @@
 
 public class AlterTableClusterStateExecutor extends DDLClusterStateTaskExecutor<AlterTableRequest> {
 
-    private static final IndicesOptions FIND_OPEN_AND_CLOSED_INDICES_IGNORE_UNAVAILABLE_AND_NON_EXISTING = IndicesOptions.fromOptions(
```

Comment:
This would also expand closed indices, while `Metadata.getIndices` will not. 
But then statements like `alter table my_table open` on closed tables should fail, why does it work?

Reply:
Good question, will check

Reply:
`Metadata.getIndices` also includes closed indices by default. To have it exclude closed indices one would have to use it like in:

https://github.com/crate/crate/blob/8220ba1ee44a5981ce6c21b4a0bbd1088fcb9b7e/server/src/main/java/io/crate/execution/ddl/tables/TransportCloseTable.java#L371-L376

```
List<Index> openIndices = currentState.metadata().getIndices(
                table,
                request.partitionValues(),
                false,
                imd -> imd.getState() == State.CLOSE ? null : imd.getIndex()
            );
```

I suspect some of the other new `getIndices` may actually be wrong because of this. Will need to double check, but if I understand your comment right, the PR here should be good as is?

Reply:
Right, thanks for checking!